docs/topics/tour/kotlin-tour-intermediate-properties.md

```diff
@@ -0,0 +1,647 @@
+        if (new > 25) {
+            println("Warning: Temperature is too high! ($old°C -> $new°C)")
+        } else {
+            println("Temperature updated: $old°C -> $new°C")
+        }
+    }
+}
+
+fun main() {
+    val thermostat = Thermostat()
+    thermostat.temperature = 22.5
+    // Temperature updated: 20.0°C -> 22.5°C
+
+    thermostat.temperature = 27.0
+    // Warning: Temperature is too high! (22.5°C -> 27.0°C)
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-tour-properties-observable"}
+
+In this example:
+
+* There is a `Thermostat` class that contains an observable property: `temperature`.
+* The `observable()` function accepts `20.0` as a parameter and uses it to initialize the property.
+* The lambda expression provided to the `observable()` function:
+  * Has three parameters:
+    * `_`, which refers to the property itself.
+    * `old`, which is the old value of the property.
+    * `new`, which is the new value of the property.
+  * Checks if the `new` parameter is greater than `25`, and depending on the result, prints a string to console.
+* The `main()` function:
+  * Creates an instance of the `Thermostat` class called `thermostat`.
+  * Updates the value of the `temperature` property of the instance to `22.5`, which triggers a print statement with a temperature update.
+  * Updates the value of the `temperature` property of the instance to `27.0`, which triggers a print statement with a warning.
+
+Observable properties are useful not only for logging and debugging purposes. You can also use them for use cases like
+updating a UI or to perform additional checks, like verifying the validity of data.
+
+For more information, see [Observable properties](delegated-properties.md#observable-properties).
+
+## Properties practice
+
+### Exercise 1 {initial-collapse-state="collapsed" collapsible="true" id="properties-exercise-1"}
+
+You manage an inventory system at a bookstore. The inventory is stored in a list where each item represents the quantity
+of a specific book. For example, `listOf(3, 0, 7, 12)` means the store has 3 copies of the first book, 0 of the second,
+7 of the third, and 12 of the fourth.
+
+Write a function called `findOutOfStockBooks()` that returns a list of indices for all the books that are out of stock.
+
+<deflist collapsible="true">
+    <def title="Hint 1">
+        Use the <a href="https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/indices.html"><code>indices</code></a> extension property from the standard library.
+    </def>
+</deflist>
+
+<deflist collapsible="true">
+    <def title="Hint 2">
+        You can use the <a href="https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/build-list.html"><code>buildList()</code></a> function to create and manage a list, instead of manually creating and returning a mutable list. The <code>buildList()</code> function uses a lambda with a receiver, which you learned about in earlier chapters.
```

Comment:
```suggestion
        You can use the <a href="https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/build-list.html"><code>buildList()</code></a> function to create and manage a list instead of manually creating and returning a mutable list. The <code>buildList()</code> function uses a lambda with a receiver, which you learned about in earlier chapters.
```